pkg/repo/scanner_test.go

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Kusari <https://www.kusari.dev/>
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"archive/tar"
+	"compress/bzip2"
+	"encoding/json"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/kusaridev/kusari-cli/api"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestScan_ArchiveFormat(t *testing.T) {
+	// Create a temporary test directory with git repo
+	testDir := t.TempDir()
+
+	// Initialize git repo
+	require.NoError(t, os.Chdir(testDir))
+	require.NoError(t, runCommand("git", "init"))
+	require.NoError(t, runCommand("git", "config", "user.email", "test@example.com"))
+	require.NoError(t, runCommand("git", "config", "user.name", "Test User"))
+
+	// Create initial commit
+	testFile := filepath.Join(testDir, "test.txt")
+	require.NoError(t, os.WriteFile(testFile, []byte("test content"), 0644))
+	require.NoError(t, runCommand("git", "add", "."))
+	require.NoError(t, runCommand("git", "commit", "-m", "initial commit"))
+
+	// Make uncommitted change
+	require.NoError(t, os.WriteFile(testFile, []byte("uncommitted change"), 0644))
+
+	uploadCalled := false
+
+	deps := &ScanDeps{
+		FileUploader: func(presignedURL, filePath string) error {
+			uploadCalled = true
+
+			// Copy the file to preserve it for inspection
+			preservedPath := filepath.Join(testDir, "preserved-archive.tar.bz2")
+			data, err := os.ReadFile(filePath)
+			require.NoError(t, err)
+			require.NoError(t, os.WriteFile(preservedPath, data, 0644))
+
+			return nil
+		},
+		PresignedURLGetter: func(apiEndpoint string, jwtToken string, filePath string) (string, error) {
+			return "https://s3.example.com/upload?epoch=1234567890", nil
+		},
+	}
+
+	// Run the scan with dependencies injection
+	err := ScanWithDeps(testDir, "HEAD", "https://platform.example.com", "https://console.example.com", false, false,
+		deps)
+	require.NoError(t, err)
+
+	// Verify upload was called
+	assert.True(t, uploadCalled, "Upload should have been called")
+
+	// Verify the archive format
+	preservedPath := filepath.Join(testDir, "preserved-archive.tar.bz2")
+	verifyArchiveFormat(t, preservedPath, filepath.Base(testDir))
+}
+
+func verifyArchiveFormat(t *testing.T, archivePath, testDir string) {
+	// Open the bz2 compressed file
+	file, err := os.Open(archivePath)
+	require.NoError(t, err)
+	defer file.Close() //nolint:errcheck
+
+	bzReader := bzip2.NewReader(file)
+	tarReader := tar.NewReader(bzReader)
+
+	foundMeta := false
+	foundPatch := false
+	foundTestFile := false
+	var metaContent api.BundleMeta
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		require.NoError(t, err)
+
+		// Check for expected files
+		switch {
+		case strings.HasSuffix(header.Name, "kusari-inspector.json"):
```

Comment:
If a file is written to `var/blah/blah/kusari-inspector.json`, will that satisfy this test? Because that's part of what was happening when I was working on the fix yesterday (and why I had to add the `-C` to the tar call.